guacamole-ext/src/main/java/org/apache/guacamole/properties/URIListGuacamoleProperty.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.properties;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.GuacamoleServerException;
+
+/**
+ * A GuacamoleProperty whose value is a List of URIs. The string value
+ * parsed to produce this list is a semicolon-delimited list.
+ * Duplicate values are ignored, as is any whitespace following delimiters.
+ * To maintain compatibility with the behavior of Java properties in general,
+ * only whitespace at the beginning of each value is ignored; trailing
+ * whitespace becomes part of the value.
+ */
+public abstract class URIListGuacamoleProperty
+        implements GuacamoleProperty<List<URI>> {
```

Comment:
Looks good, however in the context of these changes I think this should be implemented via adding collection support to the existing `URIGuacamoleProperty`.